packages/cli/src/executions/execution.service.ts

```diff
@@ -450,8 +450,19 @@ export class ExecutionService {
 			return await this.executionRepository.stopBeforeRun(execution);
 		}
 
+		// NOTE: old code
+		// NOTE: Should this actually stop the execution?
+		// Maybe this should just get the execution from the active executions,
+		// then cancel it and then let the workflow execute and workflow runner
+		// take care of the rest?
+		//if (this.activeExecutions.has(execution.id)) {
+		//	this.activeExecutions.stopExecution(execution.id);
+		//}
+
+		// NOTE: new code
+		// FIXME: Option 3
```

Comment:
Preferred solution. There is only one place that deregisters the execution:

https://github.com/n8n-io/n8n/blob/d61d563f48c06c889bef23021ab0295704ca1f50/packages/cli/src/workflow-runner.ts#L326